fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/indexes/RankIndexTest.java

```diff
@@ -745,34 +746,27 @@ void containsNullScore() throws Exception {
     }
 
     @DualPlannerTest
-    void writeOnlyRankQuery() {
-        assertThrows(RecordCoreException.class, () -> {
-            fdb = dbExtension.getDatabase();
-            try (FDBRecordContext context = openContext()) {
-                openRecordStore(context);
-                recordStore.markIndexWriteOnly("rank_by_gender").join();
-
-                // Re-open to reload state.
-                openRecordStore(context);
-                final QueryComponent filter =
-                        Query.rank(field("score").groupBy(field("gender"))).equalsValue(1L);
-                RecordQuery query = RecordQuery.newBuilder()
-                        .setRecordType("BasicRankedRecord")
-                        .setFilter(filter)
-                        .build();
-                RecordQueryPlan plan = recordStore.planQuery(query);
-                assertMatchesExactly(plan,
-                        filterPlan(
-                                typeFilterPlan(
-                                        scanPlan().where(scanComparisons(unbounded())))
-                                        .where(recordTypes(containsAll(ImmutableSet.of("BasicRankedRecord")))))
-                                .where(queryComponents(exactly(equalsObject(filter)))));
-                recordStore.executeQuery(plan)
-                        .map(rec -> TestRecordsRankProto.BasicRankedRecord.newBuilder().mergeFrom(rec.getRecord()).build()).asList().get();
-            } catch (ExecutionException e) {
-                throw e.getCause();
-            }
-        });
+    void writeOnlyRankQuery() throws Exception {
```

Comment:
This test was failing, and reworking it to isolate the assertThrows helped me debug what was wrong, and I felt like it would be good to keep it.